test/integration/features/profile.feature

```diff
@@ -2,104 +2,211 @@
-    When executing "minishift profile delete minishift --force"
-    Then exitcode should equal "1"
-     And stderr should contain
+  Examples: Wrong profile names
+    | profilename |
+    | '.'         |
+    | '-'         |
+    | '#$'        |
+    | '_VM'       |
+    | ' test'     |
+    | '@test'     |
+    | '!profile'  |
+    | 'demo@1'    |
+    | '?aaa#'     |
+    | '%pro%'     |
+    | '&foo'      |
+    | '*te$st^'   |
+    | 'foo 123'   |
+    | 'foo    '   |
+
+  Scenario Outline: As user, I can create profile with alphanumeric character including '_' and '-' in profile name
+    Given profile "minishift" has state "Does Not Exist"
+     When executing "minishift profile set <profilename>" succeeds
+     Then profile <profilename> should have state "Does Not Exist"
+     And profile <profilename> should be an active profile
+
+  Examples: Correct profile names
+    | profilename |
+    | Test-123    |
+    | profile_    |
+    | P1_name-    |
+    | vm_45       |
+    | 20          |
+    | random001z  |
+    | 00_XYZ-     |
+    | test_Pro-01 |
+    | foo         |
+
+  Scenario: As user, I can switch between existing profiles
+    Given profile "Test-123" has state "Does Not Exist"
+     When executing "minishift profile set Test-123" succeeds
+     Then profile "Test-123" should be an active profile
+     And executing "minishift profile set minishift" succeeds
+     Then profile "minishift" should be an active profile
+
+  Scenario: As user, I can start Minishift with default profile 'minishift'
+    Given profile "minishift" has state "Does Not Exist"
+    And profile "minishift" is an active profile
+     When executing "minishift start" succeeds
+     Then profile "minishift" should have state "Running"
+     And executing "minishift profile list" succeeds
+     Then stdout should match "-\s*minishift\s*Running\s*\(Active\)"
+
+  Scenario: As user, I can apply independent settings in profile 'foo'
+    Given profile "foo" has state "Does Not Exist"
+     When executing "minishift profile set foo" succeeds
+     Then profile "foo" should be an active profile
+     And executing "minishift config set memory 4096" succeeds
+     And executing "minishift config set disk-size 30g" succeeds
+     And executing "minishift config set cpus 4" succeeds
+     And executing "minishift addons enable registry-route" succeeds
+     Then executing "minishift config get memory" succeeds
+     And stdout should match "4096"
+     Then executing "minishift config get disk-size" succeeds
+     And stdout should match "30g"
+     Then executing "minishift config get cpus" succeeds
+     And stdout should match "4"
+     Then executing "minishift addons list"
+     And stdout should match "registry-route\s*: enabled\s*P\(0\)"
+
+  @minishift-only
+  Scenario: As user, I can check the config settings and add-ons list that should not be changed in profile 'minishift' from active profile 'foo' 
```

Comment:
What is the purpose of this scenario. If the previous scenario would be written in a better way, then this would be pretty much unnecessary. Also, why introducing something Minishift specific here. There is really no need for that. We should use  @minishift-only sparingly.